sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/SessionTokenMismatchRetryPolicy.java

```diff
@@ -143,7 +162,6 @@ private boolean shouldRetryLocally(SessionRetryOptions sessionRetryOptions, int
         // another attempt on the same region
         // hence to curb the retry attempts on a region,
         // compare sessionTokenMismatchRetryAttempts with max retry attempts allowed on the region - 1
-        return !(regionSwitchHint == CosmosRegionSwitchHint.REMOTE_REGION_PREFERRED
-            && sessionTokenMismatchRetryAttempts == (this.maxRetryAttemptsInCurrentRegion.get() - 1));
+        return sessionTokenMismatchRetryAttempts <= (this.maxRetryAttemptsInCurrentRegion.get() - 1);
```

Comment:
instead of maxRetryCount, I think we should check the getMinInRegionRetryTime() here, because by only checking the maxRetryCount, we can not guarantee the reads/writes will try in the local region log enough

for eample:
for write-> using 10ms as the latency 
with the default config, it will only stay in local region for 20ms, which might be too aggressive

Reply:
No -> effectiveBackoff would be 0 for default - but because 

this condition

        if (sessionRetryOptionsAccessor.getRegionSwitchHint(sessionRetryOptions) ==
                CosmosRegionSwitchHint.REMOTE_REGION_PREFERRED
            && attempt >= (this.maxRetryAttemptsInCurrentRegion.get() - 1)) {

is true it would set effectiveBackoff to the remainingMinRetryTimeInLocalRegion - so instead of 0 would use 500ms backoff time